app/src/main/java/io/appium/uiautomator2/handler/FindElement.java

```diff
@@ -38,6 +40,21 @@
 
 public class FindElement extends SafeRequestHandler {
 
+    /**
+     * java_package : type / name
+     *
+     * com.example.Test:id/enter
+     *
+     * ^[a-zA-Z_] - Java package must start with letter or underscore
+     * [a-zA-Z0-9\._]* - Java package may contain letters, numbers, periods and
+     * underscores : - : ends the package and starts the type [^\/]+ - type is
+     * made up of at least one non-/ characters \\/ - / ends the type and starts
+     * the name [\S]+$ - the name contains at least one non-space character and
+     * then the line is ended
+     */
+    static final Pattern resourceIdRegex   = Pattern
```

Comment:
I find it very helpful to add, in a comment, a link to a regular expression testing site with the regex and expected data. Not sure of the best one for Java